include/deal.II/base/mpi.h

```diff
@@ -1539,11 +1539,189 @@ namespace Utilities
     }
 
 
+    template <typename T>
+    T
+    _builtin_broadcast(const MPI_Comm    &comm,
+                       const T           &object_to_send,
+                       const unsigned int root_process)
+    {
+#  ifndef DEAL_II_WITH_MPI
+      (void)comm;
+      (void)root_process;
+      return object_to_send;
+#  else
+      MPI_Datatype datatype =
+        dealii::Utilities::MPI::internal::mpi_type_id(object_to_send);
```

Comment:
I guess there is no way to check if `mpi_type_id` is defined for that type? If that is the case, I would suggest to introduce a helper trait class, so that one can write code like: https://github.com/dealii/dealii/blob/fe164662790539d00e968fcc5efea4100a1a8dde/include/deal.II/matrix_free/fe_evaluation.h#L5252-L5253

Reply:
Isn't the existence guaranteed by the fact that the function `_builtin_broadcast` is called only by overloaded versions of `broadcast` whose argument is a built-in type?